src/cyclebane/graph.py

```diff
@@ -317,43 +241,42 @@ def map(self, node_values: MappingToArrayLike) -> Graph:
         All successors of the mapped source nodes are replaced with new nodes, one for
         each index value. The value is set as an attribute on the new source nodes
         (but not their successors).
+
+        Parameters
+        ----------
+        node_values:
+            A mapping from source node names to array-like objects. The source nodes
+            are the roots of the branches to be mapped. The array-like objects must
+            support slicing, e.g., NumPy arrays, Xarray DataArrays, Pandas DataFrames,
+            etc.
         """
-        for value_mapping in self._node_values.values():
-            if any(node in value_mapping for node in node_values):
-                raise ValueError('Node already has a value')
+        _node_values = self._node_values.merge_from_mapping(node_values)
         root_nodes = tuple(node_values.keys())
-        ndim = len(self.indices)
-        indices = {}
-        for name, values in _get_indices(node_values):
-            if name is None:
-                name = f'dim_{ndim}'
-                ndim += 1
-            indices[name] = values
-        named = tuple(indices)
-        if any([name in self.index_names for name in named]):
-            raise ValueError(
-                f'Conflicting new index names {named} with existing {self.index_names}'
-            )
-        successors = _find_successors(self.graph, root_nodes=root_nodes)
+        named = tuple(
+            idx for idx in _node_values.indices if idx not in self._node_values.indices
+        )
+
+        # Make sure root nodes exist in graph, add them if not. This choice allows for
+        # mapping, e.g., with multiple columns from a DataFrame, representing labels
+        # used later for groupby operations.
+        root_node_graph = nx.DiGraph()
+        root_node_graph.add_nodes_from(root_nodes)
+        graph = nx.compose(self.graph, root_node_graph)
```

Comment:
```suggestion 
        graph = self.graph.copy()
        graph.add_nodes_from(root_nodes)
```

This should be equivalent?